### PR TITLE
Tribe: Add error with secure settings copied to tribe

### LINKED ISSUE #32117: Tribe Node fails to start with secure settings for xpack.security
If a node is configured as a tribe node, then we automatically copy each `xpack.security.*` setting from the root settings, to each of the tribe client settings (`Security.addTribeSettings`)

However, that fails if the `xpack.security` setting is a SecureSetting as `Settings.Builder.copy` cannot copy secure settings.
The error is a somewhat cryptic
```
Caused by: java.lang.IllegalArgumentException: source key not found in the source settings
```

Given Tribe is deprecated in 6.x and removed in 7, we don't want to do much here, but we don't want a situation where nodes fail to start, and the errors are unclear.

Possible solutions:
1. Don't allow secure settings for `xpack.security.*` on tribe nodes. That is, in `addTribeSettings` fail if we find a secure setting (with a reasonable error message)
2. Check & require that the `tribe.xyz.xpack.security.*` secure setting already exist in the keystore. That is if, `xpack.security.transport.ssl.keystore.secure_password` exists in the keystore, require that `tribe.xyz.xpack.security.transport.ssl.keystore.secure_password` also exist, and fail with a reasonble error message if it does not.

### PR DESCRIPTION
**Pull request by rjernst**
This commit adds a clear error message when tribe setup attempts to copy
a secure setting into tribe settings. This behavior has never worked,
but the previous error message was very confusing, complaining about a
source key not being found later when trying to read the setting.

closes #32117

### CODE REVIEW

**Review by tvernum — APPROVED** [review]
LGTM

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java [hunk 1]
```diff
@@ -786,11 +787,16 @@ private static void addTribeSettings(Settings settings, Settings.Builder setting
             }
 
             // we passed all the checks now we need to copy in all of the x-pack security settings
-            settings.keySet().forEach(k -> {
+            SecureSettings secureSettings = Settings.builder().put(settings).getSecureSettings(); // hack to get at secure settings...
+            Set<String> secureSettingKeys = secureSettings == null ? Collections.emptySet() : secureSettings.getSettingNames();
+            for (String k : settings.keySet()) {
                 if (k.startsWith("xpack.security.")) {
+                    if (secureSettingKeys.contains(k)) {
+                        throw new IllegalArgumentException("Secure setting [" + k + "] cannot be used with tribe client node");
```
Comment (jaymode) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java]: How about building a set of invalid keys here just to be a little friendlier to a user with multiple secure settings

**Comment by rjernst** [comment]
@jaymode I pushed 80b15fe.

**Review by jaymode — APPROVED** [review]
LGTM. Thanks @rjernst

**Pull request merged**